src/types.rs

```diff
@@ -1,11 +1,42 @@
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+use std::fmt::{Debug, Error, Formatter};
+
+#[derive(Clone, Eq, Hash, PartialEq)]
 pub struct Bytes(pub Vec<u8>);
 
-impl From<Vec<u8>> for Bytes {
-    fn from(t: Vec<u8>) -> Self {
-        Bytes(t)
+impl<T> From<T> for Bytes
+where
+    T: AsRef<[u8]>,
+{
+    fn from(t: T) -> Self {
+        Bytes(Vec::from(t.as_ref()))
```

Comment:
`Bytes(t.as_ref().into())` would also work, but I'm not sure which way is preferred.

I think I like `Vec::from` slightly better because it's more clear it's a vector..